scripts/dev_changelog.py

```diff
@@ -0,0 +1,257 @@
+#!/usr/bin/env python
+"""Script to generate contributor and pull request lists
+This script generates contributor and pull request lists for release
+changelogs using Github v3 protocol. Use requires an authentication token in
+order to have sufficient bandwidth, you can get one following the directions at
+`<https://help.github.com/articles/creating-an-access-token-for-command-line-use/>_
+Don't add any scope, as the default is read access to public information. The
+token may be stored in an environment variable as you only get one chance to
+see it.
+
+Usage::
+    $ ./scripts/dev_changelog.py <token> <revision range>
+
+The output is utf8 rst.
+
+Dependencies
+------------
+- gitpython
+- pygithub
+
+Examples
+--------
+From the bash command line with $GITHUB token::
+    $ ./scripts/dev_changelog.py $GITHUB v0.3.0..v0.4.0 > 1.14.0-changelog.rst
```

Comment:
```suggestion
    $ ./scripts/dev_changelog.py $GITHUB v0.3.0..v0.4.0 > 0.4.0-changelog.rst
```

Reply:
Actually that should probably be something like
```suggestion
    $ ./scripts/dev_changelog.py $GITHUB v0.3.0..v0.4.0 -o 0.4.0-changelog.rst
```

Reply:
for now, it just prints it and redirecting it should be fine I think.

Reply:
When I used the script, it saved the changelog to `docs/source/changelog.rst`, it didn't just print it.

Reply:
@behackl  is correct
I changed the logic to write to a file instead of printing so that we could use tqdm

Reply:
Originally, it used to print now it is modified. I think the docstring must be changed

Reply:
I just did @behackl suggestion and changed the usage --- everything else could have a docstring, but it's better to spend effort first implementing @kolibril13 suggestion before documenting. Unfortunately, I have to go now